2377/2/versus/versus_battle.rb

```diff
@@ -0,0 +1,46 @@
+require_relative 'battle'
+require_relative 'rapper'
+
+# Class collects data about battles and rappers
+class VersusBattle
+  attr_reader :rappers
+  EXP = /((\w|\d|.|\s)*?)(?=против|[vV][Ss] )/
+  def initialize
+    @rappers = []
+    @files = Dir.glob('*[^.rb]')
+  end
+
+  def create_rapper_array
```

Comment:
instead of these 2 methods, you can create one method `rappers` that will return the array of rappers with battles already set and sorted.
And you iterate through every file at least twice (much more times actually)